### ISSUE TITLE
Question: Any thoughts on XMLField formatting using NewLineOnAttributes?

### ISSUE THREAD

**Issue by agehrke** [kamsar/Rainbow#12]
Hi Kam

Have you thought about using NewLineOnAttributes when formatting XML in `XmlFieldFormatter`? It should make it easier see what's changed when diff'ing commits.

Something like:

```csharp
XDocument doc = XDocument.Parse(field.Value);

XmlWriterSettings settings = new XmlWriterSettings
{
	OmitXmlDeclaration = true,
	Indent = true,
	NewLineOnAttributes = true,
};
StringBuilder result = new StringBuilder();
using (XmlWriter writer = XmlWriter.Create(result, settings))
{
	doc.WriteTo(writer);
}

return result.ToString();
```

Which will output something like:

```xml
<r xmlns:xsd="http://www.w3.org/2001/XMLSchema">
  <d
    id="{FE5D7FDF-89C0-4D99-9AA3-B5FBD009C9F3}"
    l="{0DAC6578-BC11-4B41-960A-E95F21A78D1F}">
    <p
      uid="{C4C0DC9E-3C95-45D2-BB02-EAA406B1C9A9}"
      key="main"
      md="/sitecore/layout/Placeholder Settings/Sites/Public/Home/main" />
    <r
      id="{EF37A63D-B9E1-4335-B4B1-762E4764EF5A}"
      ph="wrapper"
      uid="{6D364A99-1F9D-44F1-A849-8D8310DB24AE}" />
  </d>
</r>
```

**Comment by kamsar** [comment]
I like it. It would alter existing items on save but that is probably ok.
Easy enough to provide a legacy mode for Luddites :)

**Comment by agehrke** [comment]
Yes, it would alter existing items on save. But the Deserialize method can luckily ready any format.
Yeah, a feature flag or toggle might be needed. 

I can turn this into a pull request some time next week.

**Comment by kamsar** [comment]
Sounds great

**Comment by kamsar** [comment]
This will be fixed in the next version. A feature toggle was added to `Rainbow.config` with the option to go legacy. But don't :grinning:

**Comment by kamsar** [comment]
Fixed in 3.1.4

**Comment by agehrke** [comment]
I'm too slow :-D Great work, Kam.
